src/pages/painting/[slug].js

```diff
@@ -17,13 +17,14 @@ import Recommendation from "@/components/paintings-page/recommendation/recommend
 export default function Painting({ paintingData = {}, collectionData }) {
   // The collectionData prop has all the paintings in this collection. Use this data for the reccommended paintings section
   const { query } = useRouter();
+  const hasQuery = query.collection || "All";
```

Comment:
*nitpick
This doesn't return a boolean value, it returns a collection name. Maybe something like `collectionName`.
Nice use of the OR operator!